runtime/js/40_command.js

```diff
@@ -0,0 +1,168 @@
+        cwd: pathFromURL(cwd),
+        clearEnv,
+        env: ObjectEntries(env),
+      };
+    }
+
+    spawn(options = {}) {
+      const child = core.opSync("op_command_spawn", {
+        ...this.#args,
+        ...options,
+      });
+
+      return new Child(illegalConstructorKey, child);
+    }
+
+    async status(options = {}) {
+      return await core.opAsync("op_command_status", {
+        ...this.#args,
+        ...options,
+      });
+    }
+
+    async output() {
+      return await core.opAsync("op_command_output", this.#args);
+    }
+  }
+
+  class Child {
+    #rid;
+
+    #pid;
+    get pid() {
+      return this.#pid;
+    }
+
+    #stdin;
+    get stdin() {
+      return this.#stdin;
+    }
+
+    #stdoutRid;
+    #stdout;
+    get stdout() {
+      return this.#stdout;
+    }
+
+    #stderrRid;
+    #stderr;
+    get stderr() {
+      return this.#stderr;
+    }
+
+    constructor(key = null, {
+      rid,
+      pid,
+      stdinRid,
+      stdoutRid,
+      stderrRid,
+    } = null) {
+      if (key !== illegalConstructorKey) {
+        throw new TypeError("Illegal constructor.");
+      }
+
+      this.#rid = rid;
+      this.#pid = pid;
+
+      if (stdinRid !== undefined) {
+        this.#stdin = new WritableStream({
+          async write(chunk) {
+            await writeAll(stdinRid, chunk);
+          },
+          abort() {
+            core.close(stdinRid);
+          },
+        });
+      }
+
+      if (stdoutRid !== undefined) {
+        this.#stdoutRid = stdoutRid;
+        // TODO(crowlkats): BYOB Stream
+        this.#stdout = new ReadableStream({
+          async pull(controller) {
+            const buf = new Uint8Array(1024); // TODO(crowlkats): different number? wouldnt bee needed with byob stream
+            const res = await read(stdoutRid, buf);
+            if (res === null) {
+              core.close(stdoutRid);
+              controller.close();
+            } else {
+              controller.enqueue(buf);
+            }
+          },
+          cancel() {
+            core.close(stdoutRid);
+          },
+        });
+      }
+
+      if (stderrRid !== undefined) {
+        this.#stderrRid = stderrRid;
+        // TODO(crowlkats): BYOB Stream
+        this.#stderr = new ReadableStream({
+          async pull(controller) {
+            const buf = new Uint8Array(1024); // TODO(crowlkats): different number? wouldnt bee needed with byob stream
+            const res = await read(stderrRid, buf);
+            if (res === null) {
+              core.close(stderrRid);
+              controller.close();
+            } else {
+              controller.enqueue(buf);
+            }
+          },
+          cancel() {
+            core.close(stderrRid);
+          },
+        });
+      }
+    }
+
+    get status() {
+      return core.opSync("op_command_child_status", this.#rid);
+    }
+
+    async wait() {
+      return await core.opAsync("op_command_child_wait", this.#rid);
+    }
```

Comment:
There should by `tryWait` as well? (ie. sync API to check if process has finished)

Reply:
currently there is a `status` getter

Reply:
Okay, sounds good, it will all be clearer once there's type declaration.